client/kubernetes/hardware_finder.go

```diff
@@ -46,7 +47,7 @@ func NewFinder(logger log.Logger, k8sAPI, kubeconfig, kubeNamespace string) (*Fi
 		return nil, err
 	}
 
-	cluster, err := NewCluster(config)
+	cluster, err := NewCluster(config, kubeNamespace)
```

Comment:
```suggestion
	cluster, err := NewCluster(config, ccfg.Namespace())
```

This suggestion doesn't make any behavioral difference but its seemingly more resilient to config loading rules being tweaked in the lines above. It ensures we always pluck the namespace the config loader deduced irrespective of how it got it.

We could, alternatively, change `NewCluster` to just accept `ccfg` and have it pull the information from a single data structure (that honestly seems preferable if you're willing).